web/src/lib/data-hooks/useContentModifiedByUserData.ts

```diff
@@ -2,7 +2,8 @@ import { useUserData } from "@/lib/data-hooks/useUserData";
 import { templateEval, templateEvalWithExtraSpaceRemoval } from "@/lib/utils/helpers";
 
 export const useContentModifiedByUserData = (content: string): string => {
-  const { userData } = useUserData();
+  const { updateQueue } = useUserData();
+  const userData = updateQueue?.current();
```

Comment:
Funny that (unfortunately) this would be fixed by the refactor I'm working on right now, to use `updateQueue.current` for userData automatically